daemon/cmd/daemon_main.go

```diff
@@ -1768,6 +1771,26 @@ func runDaemon() {
 			})
 			ms.CollectStaleMapGarbage()
 			ms.RemoveDisabledMaps()
+
+			if len(d.restoredCIDRs) > 0 {
+				// Release restored CIDR identities after a grace period (default 10
+				// minutes).  Any identities actually in use will still exist after
+				// this.
+				//
+				// This grace period is not needed when running with K8s, as in that
+				// case we get here only after all state has been sychronized from
+				// K8s. If running on an external workload, or in any other case
+				// where k8s is not configured we need a grace period during which
+				// policies can be restored by importing them usign the Cilium API.
+				if !k8s.IsEnabled() {
+					time.Sleep(option.Config.IdentityRestoreGracePeriod)
+				}
```

Comment:
What do we actually need to happen before we trigger the logic below? Full policy calculation for all endpoints?

Does "k8s is synchronized" also guarantee that Cilium has processed all of the events from k8s? Or only that Cilium has received all of the resources and has begun processing them? Given the way that policy is processed asynchronously in Cilium, I'm not sure that k8s sync will provide the guarantees we need around the reference counting of the CIDR identities that are referenced here :thinking:

Reply:
@aanm Please resolve this with Joe. I made this sleep not to be done when k8s is enabled due to your change request.

Reply:
Unless otherwise noted I'll remove the !k8s condition on the sleep to make sure we do not release these identities too soon.

Reply:
Now sleeping unconditionally with this comment:
```
				// Release restored CIDR identities after a grace period (default 10
				// minutes).  Any identities actually in use will still exist after
				// this.
				//
				// This grace period is needed when running on an external workload
				// where policy synchronization is not done via k8s. Also in k8s
				// case it is prudent to allow concurrent endpoint regenerations to
				// (re-)allocate the restored identities before we release them.
```

Reply:
We can tighten this down again later if necessary.